File: PRN211_project_publish/Models/Task.cs
using System;
using System.Collections.Generic;

namespace PRN211_project_publish.Models;

public partial class Task
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public string? Description { get; set; }

    public DateTime? DueDate { get; set; }

    public bool? Status { get; set; }

    public int? ListId { get; set; }

    public DateTime? CreateDate { get; set; }

    public virtual List? List { get; set; }

    public virtual ICollection<SubTask> SubTasks { get; set; } = new List<SubTask>();

    public virtual ICollection<Tag> Tags { get; set; } = new List<Tag>();
}

File: todoapp.client/src/types/Task.ts
import { ISubTask } from "./SubTask";
import { ITag } from "./Tag";

export interface ITask {
    id: number;
    name: string;
    description: string;
    dueDate: Date;
    status: boolean;
    listId: number;
    listName: string;
    createDate: Date;
    subTasks: ISubTask[];
    tags: ITag[];
}

export interface TaskRequest {
    taskId?: number;
    name?: string;
    description?: string;
    listId: number;
    dueDate: Date;
    selectedTags: number[];
}

export interface TaskCountsDto {
    total: number;
    today: number;
    upcoming: number;
    overdue: number;
    completed: number;
}